arangod/RocksDBEngine/RocksDBTransactionMethods.h

```diff
@@ -56,7 +56,10 @@ class RocksDBTransactionMethods : public RocksDBMethods {
   virtual Result abortTransaction() = 0;
 
   // Only relevant for RocksDBTrxMethods
-  virtual Result checkIntermediateCommit() { return {}; }
+  virtual bool isIntermediateCommitNeeded() { return false; }
+  virtual Result triggerIntermediateCommit() {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
```

Comment:
Why is this a THROW and not a return Result?

I understand it should never be called, however return is handled gracefully, throw potentially not.

Reply:
If this is ever called, then this is a serious bug. Not sure if this can actually be handled gracefully in a _correct_ way. So maybe we should even use a `ADB_PROD_ASSERT(false)` (which would actually terminate the process).